### PR TITLE
Add ConnectorSetting SupportXMsEnumValues

### PR DESCRIPTION
**Pull request by LucGenetier**
Add ConnectorSettings SupportXMsEnumValues
Add management for numeric value based OptionSets

### CODE REVIEW

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.GetReturnType(Microsoft.OpenApi.Models.OpenApiOperation,Microsoft.PowerFx.Connectors.ConnectorCompatibility)

**Comment by MikeStall** [comment]
>         public string Namespace { get; }

can this be init to ensure it's immutable? 

This is what that threadingTest.cs is catching - you're introducing a shared static with mutable state


---
Refers to: src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs:27 in f0bce7d. [](commit_id = f0bce7dc971bfcdf0f7ba20c63d59bdb91d755cc, deletion_comment = False)

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs [hunk 1]
```diff
@@ -814,9 +822,9 @@ public static HttpMethod ToHttpMethod(this OperationType key)
             };
         }
 
-        public static FormulaType GetReturnType(this OpenApiOperation openApiOperation, ConnectorCompatibility compatibility)
+        public static FormulaType GetReturnType(this OpenApiOperation openApiOperation, ConnectorSettings settings)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:825]: >ConnectorSettings [](http://example.com/codeflow?start=88&length=17)

this is flagging as a public API break. Is this ok?

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:831]: I'll add this one for backward compatibility

        [Obsolete("Use a ConnectorSettings parameter instead")]
        public static FormulaType GetReturnType(this OpenApiOperation openApiOperation, ConnectorCompatibility compatibility)

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**